### PR TITLE
correctly cloning the submodelparts

### PR DESCRIPTION
**Pull request by RiccardoRossi**
this branch ensures that submodelparts are correctly cloned

### CODE REVIEW

**Comment by RiccardoRossi** [comment]
@pooyan-dadvand  i think you are the one to agree about this

File: kratos/utilities/connectivity_preserve_modeler.h [hunk 1]
```diff
@@ -129,15 +124,60 @@ class ConnectivityPreserveModeler : public Modeler
...
         KRATOS_CATCH("");
     }
+    
+protected:
+    
+    void PopulateSubModelParts(
+        ModelPart& OriginModelPart,
+        ModelPart& DestinationModelPart)
+    {
+        KRATOS_WATCH(OriginModelPart)
```
Comment (pooyan-dadvand) [inline kratos/utilities/connectivity_preserve_modeler.h]: Would you please remove the watches?

File: kratos/utilities/connectivity_preserve_modeler.h [hunk 2]
```diff
@@ -129,15 +124,60 @@ class ConnectivityPreserveModeler : public Modeler
...
+        KRATOS_WATCH(DestinationModelPart)
+        std::vector<ModelPart::NodeType::SizeType > ids;
+        for(auto part=OriginModelPart.SubModelPartsBegin(); part!=OriginModelPart.SubModelPartsEnd(); part++)
+        {
+            KRATOS_WATCH(part->Name())
```
Comment (pooyan-dadvand) [inline kratos/utilities/connectivity_preserve_modeler.h]: One more

File: kratos/utilities/connectivity_preserve_modeler.h [hunk 3]
```diff
@@ -129,15 +124,60 @@ class ConnectivityPreserveModeler : public Modeler
...
+            
+            
+            auto& destination_part = DestinationModelPart.CreateSubModelPart(part->Name());
+            
+            //add nodes
+            ids.reserve(part->Nodes().size());
+            for(auto it=part->NodesBegin(); it!=part->NodesEnd(); ++it)
+                ids.push_back(it->Id());
+            destination_part.AddNodes(ids);
```
Comment (pooyan-dadvand) [inline kratos/utilities/connectivity_preserve_modeler.h]: Is there any reason to use this version by ids? It won't be better to add it by the method accepts the iterators:

https://github.com/KratosMultiphysics/Kratos/blob/master/kratos/includes/model_part.h#L711

File: kratos/utilities/connectivity_preserve_modeler.h [hunk 4]
```diff
@@ -129,15 +124,60 @@ class ConnectivityPreserveModeler : public Modeler
...
+
+            //add conditions
+            ids.clear();
+            ids.reserve(part->Conditions().size());
+            for(auto it=part->ElementsBegin(); it!=part->ElementsEnd(); ++it)
+                ids.push_back(it->Id());
+            destination_part.AddElements(ids);
```
Comment (pooyan-dadvand) [inline kratos/utilities/connectivity_preserve_modeler.h]: Same as the nodes

File: kratos/utilities/connectivity_preserve_modeler.h [hunk 5]
```diff
@@ -129,15 +124,60 @@ class ConnectivityPreserveModeler : public Modeler
...
+            
+            //add conditions
+            ids.clear();
+            ids.reserve(part->Conditions().size());
+            for(auto it=part->ConditionsBegin(); it!=part->ConditionsEnd(); ++it)
+                ids.push_back(it->Id());
+            destination_part.AddConditions(ids);
```
Comment (pooyan-dadvand) [inline kratos/utilities/connectivity_preserve_modeler.h]: Same as the conditions

**Review by pooyan-dadvand — CHANGES_REQUESTED** [review]
In general I would suggest to remove the watches and the optional change of the add methods to avoid the ids.

File: kratos/utilities/connectivity_preserve_modeler.h (hunk 5 above)
Comment (RiccardoRossi) [reply kratos/utilities/connectivity_preserve_modeler.h]: yes you are right about iterators (and watches were leftovers)

now corrected to use iterators

**Review by pooyan-dadvand — APPROVED** [review]
👍

**Pull request merged**